src/main/java/monday/monday/parser/Parser.java

```diff
@@ -0,0 +1,140 @@
+package monday.monday.parser;
+
+import monday.monday.exception.MondayExceptions;
+import monday.monday.ui.Ui;
+import monday.task.Deadline;
+import monday.task.Event;
+import monday.task.TaskList;
+import monday.task.ToDo;
+
+
+/**
+ * Parser class is responsible for parsing user input and performing the corresponding actions.
+ */
+public class Parser {
+    /**
+     * An enumeration of available commands.
+     */
+    private enum Command {
+        BYE,
+        LIST,
+        MARK,
+        UNMARK,
+        TODO,
+        DEADLINE,
+        EVENT,
+        DELETE,
+        FIND,
+        HI
+    }
+
+    /**
+     * Parses the user input and performs the corresponding action.
+     *
+     * @param userInput the user input to be parsed
+     * @param taskList the TaskList object to perform actions on
+     * @return response from chatbot
+     * @throws MondayExceptions if there are errors related to the Monday application
+     * @throws IllegalArgumentException if the user input is in the wrong format
+     */
+    public static String mondayParser(String userInput, TaskList taskList) throws MondayExceptions {
```

Comment:
like the reviewer before me has mentioned, this isn't phrased as a verb :P

I'm a lot less sure of this, but:
do you think it's a good idea to make this method throw `NumberFormattingException` as well? Your code handles it properly, but I thought it might add a bit of clarity to the types of exceptions expected from this method. 
To some extent, this ambiguity is inherent for the Exception design pattern in general, though, so feel free to disagree.

Reply:
You have a point there on the throwing of NumberFormattingException, I will look into adding that too. Thank you :>.